pallets/order-book/src/storage_data_layer.rs

```diff
@@ -92,7 +92,7 @@ impl<T: Config> StorageDataLayer<T> {
             .get(price)
             .map(|x| *x)
             .unwrap_or_default()
-            .checked_add(*value)
+            .checked_add(value)
             .ok_or(())?;
```

Comment:
I think you can replace `Result<T, ()>` with `Option<T>` everywhere. It will make checked ops usage easier

For example
```rust
let x = a.checked_add(b)?.checked_sub(c)?;
```

Reply:
We have one approach here that returns `Result<T, ()>` for different functions. Not all of them use checked math operations. Moreover, we have some cases like `Result<(), ()>` for functions that returns nothing, but something can go wrong. If we replace it on `Option<()>` it became more discuraged and not such clear as `Result`